pkg/kubectl/rollout_status.go

```diff
@@ -135,14 +135,14 @@ func (s *StatefulSetStatusViewer) Status(obj runtime.Unstructured, revision int6
 		return fmt.Sprintf("Waiting for %d pods to be ready...\n", *sts.Spec.Replicas-sts.Status.ReadyReplicas), false, nil
 	}
 	if sts.Spec.UpdateStrategy.Type == appsv1.RollingUpdateStatefulSetStrategyType && sts.Spec.UpdateStrategy.RollingUpdate != nil {
-		if sts.Spec.Replicas != nil && sts.Spec.UpdateStrategy.RollingUpdate.Partition != nil {
+		if sts.Spec.Replicas != nil && *sts.Spec.UpdateStrategy.RollingUpdate.Partition > 0 {
```

Comment:
tese case adds done @smarterclayton

Reply:
@zhangmingld I think there should be 2 checks because nil and 0 are valid situations: https://github.com/kubernetes/kubernetes/issues/72212#issuecomment-468556805